src/vulkan-renderer/world/collision.cpp

```diff
@@ -0,0 +1,184 @@
+                               cube_corner{"left back botom", adjust_coordinates({-1.0f, 1.0f, -1.0f})},
+                               cube_corner{"right back bottom", adjust_coordinates({1.0f, 1.0f, -1.0f})},
+                               cube_corner{"left front top", adjust_coordinates({-1.0f, -1.0f, 1.0f})},
+                               cube_corner{"right front top", adjust_coordinates({1.0f, -1.0f, 1.0f})},
+                               cube_corner{"left front bottom", adjust_coordinates({-1.0f, -1.0f, -1.0f})},
+                               cube_corner{"right front bottom", adjust_coordinates({1.0f, -1.0f, -1.0f})}};
+
+    using corner_on_face_index = std::array<std::size_t, 4>;
+
+    // These indices specify which 4 points describe the corners on the given face.
+    static constexpr std::array CORNERS_ON_FACE_INDICES{
+        corner_on_face_index{0, 2, 4, 6}, // left
+        corner_on_face_index{1, 3, 5, 7}, // right
+        corner_on_face_index{4, 5, 6, 6}, // front
+        corner_on_face_index{0, 1, 2, 3}, // back
+        corner_on_face_index{0, 1, 4, 5}, // top
+        corner_on_face_index{2, 3, 6, 7}  // bottom
+    };
+
+    // The coordinates of the center of every edge of the cube.
+    std::array cube_edges = {cube_edge{"left top", adjust_coordinates({-1.0f, 0.0f, 1.0f})},
+                             cube_edge{"left front", adjust_coordinates({-1.0f, 1.0f, 0.0f})},
+                             cube_edge{"left bottom", adjust_coordinates({-1.0f, 0.0f, -1.0f})},
+                             cube_edge{"left back", adjust_coordinates({-1.0f, -1.0f, 0.0f})},
+                             cube_edge{"right top", adjust_coordinates({1.0f, 0.0f, 1.0f})},
+                             cube_edge{"right front", adjust_coordinates({1.0f, 1.0f, 0.0f})},
+                             cube_edge{"right bottom", adjust_coordinates({1.0f, 0.0f, -1.0f})},
+                             cube_edge{"right back", adjust_coordinates({1.0f, -1.0f, 0.0f})},
+                             cube_edge{"middle bottom front", adjust_coordinates({0.0f, -1.0f, -1.0f})},
+                             cube_edge{"middle bottom back", adjust_coordinates({0.0f, 1.0f, -1.0f})},
+                             cube_edge{"middle top front", adjust_coordinates({0.0f, -1.0f, 1.0f})},
+                             cube_edge{"middle top back", adjust_coordinates({0.0f, 1.0f, 1.0f})}};
+
+    using edge_on_face_index = std::array<std::size_t, 4>;
+
+    // These indices specify which 4 edges are associated with a given face.
+    static constexpr std::array EDGE_ON_FACE_INDICES{
+        edge_on_face_index{0, 1, 2, 3},   // left
+        edge_on_face_index{4, 5, 6, 7},   // right
+        edge_on_face_index{1, 5, 8, 11},  // front
+        edge_on_face_index{3, 7, 9, 11},  // back
+        edge_on_face_index{0, 4, 10, 11}, // top
+        edge_on_face_index{2, 6, 8, 9}    // bottom
```

Comment:
Where is left, right, front, etc.?
Also this does not seem to be as in our documentation.

Reply:
I will check this and make it consistent.

Reply:
The order of bounding box corners is consistent with the octree corner order.

Reply:
Edges are not even part of octree docs.

Reply:
The order of axis of the cube's bounding box is the same as for octrees, but the center is `(0,0,0)`.

Reply:
I think you didn't added the edges to the documentation?
I wrote the order on discord, not sure if you wanted to use this or not.

Reply:
I added the edges to the documentation.